Utils/SceneUtils.cs

```diff
@@ -250,6 +251,67 @@ public static void GetActors()
             }
         }
 
+        public static void UpdateSceneByNumber(int SceneNumber)
+        {
+            for (int i = Math.Min(RomData.SceneList.Count, SceneNumber); i >= 0; i--)
+            {
+                if (RomData.SceneList[i].Number == SceneNumber)
+                {
+                    UpdateScene(RomData.SceneList[i]);
+                }
+            }
+        }
+
+        public static void WriteShuffledDungeonChests(int[][] ChestShuffle)
+        {
+            SceneUtils.ReadSceneTable();
+            SceneUtils.GetMaps();
+            SceneUtils.GetMapHeaders();
+            SceneUtils.GetActors();
+            // WF, SH, GB, ST, IST
+            int[] DungeonScene = new int[] { 0x1B, 0x21, 0x49, 0x16, 0x18 };
+            // WF
+            // spinning flower fairy, first room fairy, Compass, Small Key, Map, Bow, BK, dark room fairy
+            // SH
+            // Compass Fairy, Compass, Freezard Bridge Key, Push block fairy, push block key
+            // Lava Fairy, BK, Deku Flower Fairy, Freezard Stalagmite Fairy, Fire Arrows
+            // Ice Stalactite Fairy, Ice Stalactite Key, Invisible Stair Platform Fairy, Map
+            // GB
+            // Map, Bio Baba Fairy, BK, Key, Compass, Ice Arrows, Water Wheel Fairy, Hidden Torch Fairy, 
+            // Seesaw Fairy, Yellow/Green Reservoir Fairy, First Room Fairy
+            // ST ... oh god ... I'll suffix inverted temple fairy chests with (i)
+            // Remains Platform (i), Lower First Room, Upper First Room, Eyegore, Eyegore Sunblock,
+            // Giant, Eyegore Key, Elegy Key, Wind Elevator (i), Frozen Eye Switch (i), Underwater Sun Switch [have to fact check if this is the one tied to the sun switch]
+            // Beetle Guarded Key, Compass, Wizzrobe (i), Wizzrobe Key, Wizzrobe, Map [assuming that's the one in the back]
+            // Mirror Shield Sun Switch, Sunblock Room, Post Light Arrows Room, Fire Ring, Rupee Nook, Light Arrows
+            int DungeonSceneNumber;
+            for (int d = 0; d < DungeonScene.Length; d++)
+            {
+                DungeonSceneNumber = DungeonScene[d];
+                List<Actor> TempleChests = ActorUtils.GetSceneActorsByNumber(DungeonSceneNumber, 0x0006);
+                List<short> ChestContents = new List<short>();
+                ushort GetItemMask = 0xFF0, ChestFlagMask = 0xF;
+                int ChestFlagBits = 4;
+                int ChestTypeBits = 12;
```

Comment:
maybe it's just that I've been doing this stuff for years now, but it seems overkill to define variables for the flags. Additionally, GetItemMask and ChestFlagMask appear to be wrong: according to my documentation they should be 0x0FE0 and 0x001F respectively.

Reply:
I am relatively new to bit masking, but again this was introduced so I could experiment a bit. It seemed like DB altered the functioning of chests to allow expansion of the get-item table beyond 128 entries to be placed in chests. Chests were getting their Type changed when I was only giving the get-item portion 8 bits, 

I suspect this will break Stone Tower significantly tho, since there is a good chance Stone Tower uses more than 16 Chest Flags. And if I'm right about that, I can return it back to the established documentation.